cmd/compgen/README.md

```diff
@@ -59,7 +67,8 @@ to generate aditional code for the component.
 See the [README](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/README.md)
 for details on how to use mdatagen.
 
-You can run mdatagen by running `make gen`.
+Assuming you have uncommented the include of Makefile.Common in the generated
+receiver, you can run mdatagen by with `make mdatagen`.
```

Comment:
Run `make generate` instead

Reply:
This is a good callout actually but I think it warrants a little more explaining.

`make gen` is defined in the [Makefile.Common](https://github.com/liatrio/liatrio-otel-collector/blob/dc597d2316477c208f98903403f1d5f8c7ed5abf/Makefile.Common#L62) and does what I explained above. `make generate` is defined in the [Makefile](https://github.com/liatrio/liatrio-otel-collector/blob/dc597d2316477c208f98903403f1d5f8c7ed5abf/Makefile#L53) and basically runs `make gen` AKA `go generate ./...` on every location with a discovered [go.mod](https://github.com/liatrio/liatrio-otel-collector/blob/dc597d2316477c208f98903403f1d5f8c7ed5abf/Makefile#L8) file.

The basic idea is to run `make generate` after every code change, _always_. It'll either do nothing or create necessary changes. The documentation should reflect this

Reply:
Should the compgen Makefile include the root Makefile and not Makefile.common?

Reply:
`mdatagen` is removed but currently using the `make gen` target from Makefile.Common since that is what the generated Makefile imports

Reply:
@acramsay  is correct, it would be `make gen`. It should include `Makefile.Common` not `Makefile`. I presume that this is instructions for the very first time you populate this, because once you have your component, you're going to be using the w/e rules exist at the root. Additionally, when you run `make checks` at the root, it's going to run through all of this, and the workflows will catch it too.